pkg/user/tx_client_test.go

```diff
@@ -447,3 +452,241 @@ func setupEstimatorService(t *testing.T) *mockEstimatorServer {
+					t.Error("Mock service 4 should not have been called")
+					return nil, errors.New("err4")
+				}}
+				conn1, stop1 := grpctest.StartMockServer(t, mockSvc1)
+				conn2, stop2 := grpctest.StartMockServer(t, mockSvc2)
+				conn3, stop3 := grpctest.StartMockServer(t, mockSvc3)
+				conn4, stop4 := grpctest.StartMockServer(t, mockSvc4)
+				return []*grpctest.MockTxService{mockSvc1, mockSvc2, mockSvc3, mockSvc4}, []*grpc.ClientConn{conn1, conn2, conn3, conn4}, []func(){stop1, stop2, stop3, stop4}
+			},
+			expectedErr: nil,
+		},
+		{ // Non-Zero Code Failure
+			setupMocks: func(t *testing.T) ([]*grpctest.MockTxService, []*grpc.ClientConn, []func()) {
+				mockSvc1 := &grpctest.MockTxService{BroadcastHandler: func(ctx context.Context, req *sdktx.BroadcastTxRequest) (*sdktx.BroadcastTxResponse, error) {
+					resp := &sdk.TxResponse{Code: 5, TxHash: "HASH_FAIL", RawLog: errInsufficientFunds.Error()}
+					return &sdktx.BroadcastTxResponse{TxResponse: resp}, nil
+				}}
+				conn1, stop1 := grpctest.StartMockServer(t, mockSvc1)
+				return []*grpctest.MockTxService{mockSvc1}, []*grpc.ClientConn{conn1}, []func(){stop1}
+			},
+			expectedErr: &user.BroadcastTxError{}, // Expect specific error type
+		},
+	}
+
+	for i, tc := range testCases {
+		name := fmt.Sprintf("BroadcastTestCase%d", i) // Simple naming
+		t.Run(name, func(t *testing.T) {
+			_, conns, stops := tc.setupMocks(t)
+			for _, stop := range stops {
+				defer stop()
+			}
+			require.NotEmpty(t, conns, "Need at least one connection for broadcast test client")
+
+			primaryConn := conns[0]
+			otherConns := conns[1:]
+
+			// Create a temporary TxClient for this test case using the suite's signer/config
+			// but the mock connections.
+			tempTxClient, err := user.NewTxClient(
+				suite.encCfg.Codec,
+				suite.txClient.Signer(), // Reuse signer from main client
+				primaryConn,             // Use mock primary connection
+				suite.encCfg.InterfaceRegistry,
+				user.WithAdditionalCoreEndpoints(otherConns), // Use mock secondaries
+				user.WithDefaultAccount(suite.txClient.DefaultAccountName()), // Use main client's default account
+			)
+			require.NoError(t, err, "Failed to create temporary TxClient for test case %d", i)
+
+			var ctx context.Context
+			var cancel context.CancelFunc
+			if errors.Is(tc.expectedErr, context.DeadlineExceeded) {
+				ctx, cancel = context.WithTimeout(context.Background(), 50*time.Millisecond) // Short timeout for deadline test
+			} else {
+				ctx, cancel = context.WithTimeout(context.Background(), 10*time.Second) // General timeout
+			}
+			defer cancel()
+
+			resp, err := tempTxClient.BroadcastTx(ctx, []sdk.Msg{defaultMsg}, defaultOpts...)
+
+			if tc.expectedErr == nil {
+				require.NoError(t, err, "Expected success, but got error: %v", err)
+				require.NotNil(t, resp, "Expected non-nil response on success")
+				require.Equal(t, abci.CodeTypeOK, resp.Code, "Expected CodeTypeOK on success")
+			} else {
```

Comment:
please don't nest this logic, just move on to next case if tx.expectedErr == nil

Reply:
done